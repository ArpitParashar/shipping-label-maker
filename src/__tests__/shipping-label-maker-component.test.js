import React from 'react';
import ShippingLabelMakerComponent from '../features/shipping-label-maker/shipping-label-maker-component';
import ShippingLabel from '../features/shipping-label-maker/shipping-label';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

let realUseContext;
let useContextMock;
// Setup mock
beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

describe('Shipping Label Maker Component', () => {
  it('should render using context', () => {
    useContextMock.mockReturnValue([]);
    const element = new ShallowRenderer().render(
      <ShippingLabelMakerComponent />
    );
    expect(element.props.children).toMatchSnapshot();
  });

  it('should change based on on complete passed by the wizard component', () => {
    useContextMock.mockReturnValue([]);
    const parentElement = shallow(
        <ShippingLabelMakerComponent />
      );
    const endStep = parentElement.find('#testComplete');
    endStep.simulate('click');
    expect(parentElement.find(ShippingLabel)).toHaveLength(1);
  });
});
