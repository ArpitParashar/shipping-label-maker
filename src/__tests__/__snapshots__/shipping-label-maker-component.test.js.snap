exports[`Shipping Label Maker Component should render using context 1`] = `
Array [
  <Unknown
    className="hide"
    id="testComplete"
    onClick={[Function]} />,
  <WizardComponent
    onComplete={[Function]}
    steps={
        Array [
            Object {
              "component": <Unknown />,
              "step": 1,
            },
            Object {
              "component": <Unknown />,
              "step": 2,
            },
            Object {
              "component": <Unknown />,
              "step": 3,
            },
            Object {
              "component": <Unknown />,
              "step": 4,
            },
            Object {
              "component": <Unknown />,
              "step": 5,
            },
            Object {
              "component": <Unknown />,
              "step": 6,
            },
          ]
    } />,
]
`;
